### DISCUSSION TITLE
Şapka mantık sorusu

### DISCUSSION THREAD

**Discussion by sonsuzus** [sonsuzus/sonsuzus#281] [Zeka Sorusu] [answered]
3 kişiye; 3 kırmızı 2 beyaz şapka gösterildikten sonra, kafalarına giydiriyorlar. Durum şu yalnız, A kişisi hiç kimseyi göremiyor, B sadece önündeki A yı görebiliyor. C ise A yı ve B yi görebiliyor. Yani tek sıra halinde dizilmişler. Ve herkesten kafalarındaki şapkayı bilmeleri isteniyor. Konuşmalar şu şekilde gelişiyor.
A: Bilmiyorum
B: Bilmiyorum
C: Bilmiyorum
A: Bilmiyorum
B: Bilmiyorum
C: Bilmiyorum
A: Biliyorum, diyerek kafasındaki şapkanın rengini söyler.
Soru: A nın kafasında ne renk şapka vardır ve bunu nasıl bilmiştir?

**Comment by yigidi** [comment, marked as answer]
Öncelikle, B ve C'nin cevaplarına bakalım:

B, kendi kafasındaki şapkanın rengini bilemiyor. Bu durumda, A'nın kafasındaki şapkanın rengini bilebilmesi için önündeki A'nın kafasındaki şapkanın rengini bilmeli olmalıdır. Ancak B, A'nın kafasındaki şapkanın rengini bilemiyor, bu yüzden A'nın kafasındaki şapkanın rengini çıkaramaz.

C, A'nın kafasındaki şapkanın rengini bilemiyor. Bu durumda, A'nın kafasındaki şapkanın rengini bilebilmesi için hem önündeki B'nin kafasındaki şapkanın rengini bilmeli hem de B'nin A'yı görebildiği için A'nın kafasındaki şapkanın rengini çıkarmalıdır. Ancak C, A'nın kafasındaki şapkanın rengini bilemiyor, bu yüzden A'nın kafasındaki şapkanın rengini çıkaramaz.

Sonuç olarak, A'nın kafasındaki şapkanın rengini bilemediği ve diğerleri de bilemediği bilinmektedir. Ancak, dördüncü seferde A "Biliyorum" der. Bu durumda, A'nın kafasındaki şapkanın rengini nasıl bildiği sorgulanır.

A, "Biliyorum" diyerek kendi kafasındaki şapkanın rengini söyler. Burada önemli olan nokta, A'nın diğerlerinin cevaplarını dikkatlice dinlemesidir. Eğer A kendi kafasındaki şapkanın rengini bilemiyorsa, üçüncü seferde C "Bilmiyorum" yerine "Beyaz" derdi. Çünkü C hem A'yı hem de B'yi görüyor ve B'nin de kendi kafasındaki şapkanın rengini bilemediğini biliyor. Ancak C "Bilmiyorum" der. Bu durumda, A kendi kafasındaki şapkanın rengini belirleyebilir.

A, C'nin "Bilmiyorum" demesinden sonra kendi kafasındaki şapkanın rengini belirlemiş olmalıdır. Bu durumda, A'nın kafasındaki şapkanın rengi beyazdır.
